src/app/Classes/LogViewer.php

```diff
@@ -0,0 +1,200 @@
+        'warning'   => 'warning',
+        'error'     => 'warning',
+        'critical'  => 'warning',
+        'alert'     => 'warning',
+        'emergency' => 'warning',
+        'processed' => 'info',
+    ];
+
+    /**
+     * Log levels that are used.
+     *
+     * @var array
+     */
+    private static $log_levels = [
+        'emergency',
+        'alert',
+        'critical',
+        'error',
+        'warning',
+        'notice',
+        'info',
+        'debug',
+        'processed',
+    ];
+
+    const MAX_FILE_SIZE = 52428800;
+
+    /**
+     * @param string $file
+     * @throws \Exception
+     */
+    public static function setFile($file)
+    {
+        $file = self::pathToLogFile($file);
+
+        if (app('files')->exists($file)) {
+            self::$file = $file;
+        }
+    }
+
+    /**
+     * @param string $file
+     *
+     * @throws \Exception
+     *
+     * @return string
+     */
+    public static function pathToLogFile($file)
+    {
+        $logsPath = storage_path('logs');
+
+        if (app('files')->exists($file)) { // try the absolute path
+            return $file;
+        }
+
+        $file = $logsPath.'/'.$file;
+
+        // check if requested file is really in the logs directory
+        if (dirname($file) !== $logsPath) {
+            throw new \Exception('No such log file');
+        }
+
+        return $file;
+    }
+
+    /**
+     * @return string
+     */
+    public static function getFileName()
+    {
+        return basename(self::$file);
+    }
+
+    /**
+     * @return array
+     * @throws \Illuminate\Container\EntryNotFoundException
+     */
+    public static function all()
+    {
+        $log = [];
+
+        $pattern = '/\[\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\].*/';
+
+        if (!self::$file) {
+            $log_file = self::getFiles();
+            if (!count($log_file)) {
+                return [];
+            }
+            self::$file = $log_file[0];
+        }
+
+        if (app('files')->size(self::$file) > self::MAX_FILE_SIZE) {
+            return;
+        }
+
+        $file = app('files')->get(self::$file);
+
+        preg_match_all($pattern, $file, $headings);
+
+        if (!is_array($headings)) {
+            return $log;
+        }
+
+        $log_data = preg_split($pattern, $file);
+
+        if ($log_data[0] < 1) {
+            array_shift($log_data);
+        }
+
+        foreach ($headings as $h) {
+            for ($i = 0, $j = count($h); $i < $j; $i++) {
+                foreach (self::$log_levels as $level) {
+                    if (strpos(strtolower($h[$i]), '.'.$level) || strpos(strtolower($h[$i]), $level.':')) {
+                        preg_match('/^\[(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2})\](?:.*?(\w+)\.|.*?)'.$level.': (.*?)( in .*?:[0-9]+)?$/i', $h[$i], $current);
+                        if (!isset($current[3])) {
+                            continue;
+                        }
+
+                        $log[] = [
```

Comment:
Right...so this explains what the indexes from $current are. As noted above, this would read more nicely if we got associated arrays.